### PR TITLE
Change and clarify imod_coupler release process

### PR DESCRIPTION
**Pull request by Huite**

### CODE REVIEW

File: docs/coupler_release.qmd [hunk 1]
```diff
@@ -21,22 +21,30 @@ Before starting the release process, ensure that all tests are passing and that
 
 ## Update version numbers of the components as needed
 
-The components have their own version number which generally uses [Semantic Versioning](https://semver.org/), with minor version signifying a breaking release for pre-1.0 versions. 
-If a component did not change at all between releases, the version number can stay the same.
+The components have a version number which generally uses [Semantic Versioning](https://semver.org/), with minor version signifying a breaking release for pre-1.0 versions.
```
Comment (Hofer-Julian) [inline docs/coupler_release.qmd:24]: Not anymore when we switch everything to CalVer :)

File: docs/coupler_release.qmd [hunk 2]
```diff
@@ -21,22 +21,30 @@ Before starting the release process, ensure that all tests are passing and that
...
-If a component did not change at all between releases, the version number can stay the same.
+The components have a version number which generally uses [Semantic Versioning](https://semver.org/), with minor version signifying a breaking release for pre-1.0 versions. 
+The components are only guaranteed to work together if they are built at the same time.
+To clearly communicate this, component versions should have synchronized version numbers.
+This means that both version numbers should be updated even if only one component has been changed.
 
 Now submit a pull request which updates the version numbers of the components as needed.
 
-- Remove the `-dev` suffix from the version number and increase the version number in `__init__.py` of `imod_coupler` and `pre-processing/primod`
+- Create a new branch and switch to it, for example: `git checkout -b new-release`.
```
Comment (Hofer-Julian) [inline docs/coupler_release.qmd:31]:
```suggestion
- Create a new branch and switch to it, for example: `git switch --create new-release`.
```

File: docs/coupler_release.qmd [hunk 3]
```diff
@@ -21,22 +21,30 @@ Before starting the release process, ensure that all tests are passing and that
...
+- Remove the `-dev` postfix and increase the version number in `__init__.py` of `imod_coupler` and `pre-processing/primod`
```
Comment (Hofer-Julian) [inline docs/coupler_release.qmd:32]:
```suggestion
```

Let's drop that here and the section further below. (and open a PR on imod_coupler to remove the prefix)

File: docs/coupler_release.qmd [hunk 4]
```diff
@@ -21,22 +21,30 @@ Before starting the release process, ensure that all tests are passing and that
...
+- Create a Pull Request
+- Merge the Pull Request into main
 
-For example: `__version__ = "0.10.0-dev"` -> `__version__ = "0.11.0"`
+## Create a new release
 
-- Create a new commit with the updated version number,
-and push to remote
+Create a new tag like `v2023.8.0`, filling in the current year, month and a sequential "MICRO" number.
+This follows `vYYYY.MM.MICRO` from [calver](https://calver.org/).
```
Comment (Hofer-Julian) [inline docs/coupler_release.qmd:42]: Please copy exactly the same thing as we have in Ribasim nowadays

File: docs/coupler_release.qmd [hunk 5]
```diff
@@ -21,22 +21,30 @@ Before starting the release process, ensure that all tests are passing and that
...
 
-## Create a new release
+::: {.callout-note}
+Note the lack of a trailing zero on the month (`MM` verus `0M`) for parsing reasons.
+:::
```
Comment (Hofer-Julian) [inline docs/coupler_release.qmd:46]:
```suggestion
```

**Review by Hofer-Julian — COMMENTED** [review]
I had a few comments :)

File: docs/coupler_release.qmd (hunk 4 above)
Comment (Huite) [reply docs/coupler_release.qmd]: Done

File: docs/coupler_release.qmd [hunk 6]
```diff
@@ -21,22 +21,31 @@ Before starting the release process, ensure that all tests are passing and that
...
 ## Update version numbers of the components as needed
 
-The components have their own version number which generally uses [Semantic Versioning](https://semver.org/), with minor version signifying a breaking release for pre-1.0 versions. 
-If a component did not change at all between releases, the version number can stay the same.
+Determine the new version number like `2023.1.0`, filling in the current year, a bumped `MINOR` number for normal releases and a bumped `MICRO` number for non-breaking, hotfix releases.
+This follows `YYYY.MINOR.MICRO` from [calver](https://calver.org/).
 
-Now submit a pull request which updates the version numbers of the components as needed.
+The components are only guaranteed to work together if they are built at the same time.
+To clearly communicate this, component versions should have synchronized version numbers.
```
Comment (Hofer-Julian) [inline docs/coupler_release.qmd:28]:
```suggestion
To clearly communicate this, component versions need to have synchronized version numbers.
```

**Review by Hofer-Julian — COMMENTED** [review]
One minor comment, apart from that ready to go

**Pull request merged**